sqlmodel/main.py

```diff
@@ -643,7 +704,11 @@ def get_column_from_field(field: Any) -> Column:  # type: ignore
         unique = False
     if foreign_key:
         assert isinstance(foreign_key, str)
-        args.append(ForeignKey(foreign_key))
+        foreign_key_obj = ForeignKey(foreign_key)
+        ondelete = getattr(field_info, "ondelete", Undefined)
+        if isinstance(ondelete, str):
+            foreign_key_obj.ondelete = ondelete
```

Comment:
Here we could better use the same trick of creating args and kwargs outside before creating the `ForeignKey` object and then create it once with everything. Instead of setting the `foreign_key_obj.ondelete = ondelete` afterwards.

Just because SQLAlchemy could do some additional magic apart from setting the attribute when instantiating the object. It currently doesn't do anything, but it wouldn't be surprising if it did at some point, so better be safe. :nerd_face:

Reply:
Done